packages/neon/neon_talk/lib/src/widgets/unread_indicator.dart

```diff
@@ -0,0 +1,56 @@
+import 'package:flutter/material.dart';
+import 'package:nextcloud/spreed.dart' as spreed;
+
+/// Displays the number of unread messages and whether the user was mentioned for a given [room].
+class TalkUnreadIndicator extends StatelessWidget {
+  /// Creates a new Talk unread indicator.
+  const TalkUnreadIndicator({
+    required this.room,
+    super.key,
+  });
+
+  /// The room that the indicator will display unread messages and mentions for.
+  final spreed.Room room;
+
+  @override
+  Widget build(BuildContext context) {
+    assert(room.unreadMessages > 0, 'Need at least on unread message');
+
+    final colorScheme = Theme.of(context).colorScheme;
+
+    final highlight = room.unreadMention || spreed.RoomType.fromValue(room.type).isSingleUser;
+    final backgroundColor = highlight ? colorScheme.primaryContainer : colorScheme.background;
+    final textColor = highlight ? colorScheme.onPrimaryContainer : colorScheme.onBackground;
```

Comment:
back to the design :sweat_smile: 
1. `primaryContainer` looks good in light mode but I really feal shouted at when looking at it in dark mode. I think `secondaryContainer` would be better. It looks the same in light mode but better when in dark.
2. I'm not sure why we should distinguish between highlighted and not highlighted messaged. For me all messages are equally important. Does this make more sense in a corporate environment where nc talk is most often used?
3. regarding the `@` in room mentions. I agree that this should probably be configurable. But let's do it in a separate PR. I think that this (and probably other stuff) should be stored on a per room basis but I don't know how our current storage would handle this.
Maybe we should somehow expose one persistence per app that can be used for such purposes?

Reply:
About 2: Yeah I think the reasoning is that you can get quite a lot of messages when you are in a lot of groups and therefore messages that are considered more important are highlighted. We can also chose to not follow this, but I don't see why we shouldn't adopt it too.

Reply:
About 1: I see your point, but I think `secondaryContainer` is very ugly in dark mode. I would like to just stick with `primaryContainer`.

Reply:
About 3: Yeah we'll need something, but we can do it in a later PR.